### PR TITLE
feat: modernize setup project

### PR DESCRIPTION
**Pull request by huuduy005**
## What problem you’re solving

This PR tries to modernize the setup project and improve process build.
Hope to improve our Developer Experience.

## Any tests you wrote

No update.  I just updated the setup config.

## Changes:

* Adopt [tsup](https://github.com/egoist/tsup) to improve process build.
* Update package.json - [Conditional exports](https://nodejs.org/dist/latest-v18.x/docs/api/packages.html#conditional-exports), enrich metadata.
* Setup [prettier](https://prettier.io/).
* Correct config tsconfig.json, jest.config.js, .eslintrc.js, .eslintignore.
* Enrich .gitignore.
* Distributes as ESM, CJS.
* Replace ts-jest with [@swc/jest](https://swc.rs/docs/usage/jest).

### CODE REVIEW

File: pnpm-lock.yaml
Comment (qtdan) [inline pnpm-lock.yaml:1]: do we need this file?

**Review by qtdan — COMMENTED** [review]
Thanks for the PR.
Please also help to update [contribution guide](CONTRIBUTING.md) about your changes

File: pnpm-lock.yaml
Comment (huuduy005) [reply pnpm-lock.yaml:1]: Sorry, my environment uses [pnpm](https://pnpm.io/) as package manager, an alternative to npm.
So pnpm-lock.yaml acts like package-lock.json
Best practice setup refers to committing this file to source repositories. If we use npm so I can regenerate package-lock.json.
Ref: https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json

**Comment by huuduy005** [comment]
> Thanks for the PR. Please also help to update [contribution guide](CONTRIBUTING.md) about your changes

I just updated the setup configuration, don't change the way to contribute. What information do you want me to update?

**Comment by qtdan** [comment]
> > Thanks for the PR. Please also help to update [contribution guide](CONTRIBUTING.md) about your changes
> 
> I just updated the setup configuration, don't change the way to contribute. What information do you want me to update?

I see you have some changes in package.json for build config
```
"generate": "sh scripts/generate.sh",
```

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,145 @@
...
+.env.production.local
+.env.local
+
+# Serverless directories
+.serverless/
+
+# FuseBox cache
+.fusebox/
+
+# DynamoDB Local files
```
Comment (longxice) [inline .gitignore:99]: @huuduy005 h
 I think this SDK that doesn't use database so DynamoDB can be removed?

Comment (anhldbk) [reply .gitignore:99]: @longxice I think Duy is using [this extension](https://marketplace.visualstudio.com/items?itemName=piotrpalarz.vscode-gitignore-generator) to generate .gitignore files. It's ok for me to see bloated lines here :D

File: jest.config.js [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
   testEnvironment: "node",
-  testMatch: [
-    "**/src/zalopay/__tests__/**/*.test.ts"
-  ],
+  testTimeout: 60_000,
+  transform: {
+    "^.+\\.ts$": "@swc/jest"
+  },
+  moduleFileExtensions: ["ts", "js", "json"],
+  testMatch: ["**/*.test.ts"]
```
Comment (longxice) [inline jest.config.js:10]: @leonthaibao , @noradomi I see current config that jest executor only run the test files in the "**/src/zalopay/__tests__/" . This change impacts to us?

File: .gitignore (hunk 1 above)
Comment (huuduy005) [reply .gitignore:99]: I use [.​ignore](https://plugins.jetbrains.com/plugin/7495--ignore) to generate file .gitignore.
It is based on a collection [gitignore](https://github.com/github/gitignore) so I think it's ok @longxice

File: jest.config.js (hunk 2 above)
Comment (huuduy005) [reply jest.config.js:10]: @longxice I change the `testMatch` to match more test files, not only in `/src/zalopay/tests/`. So it's ok.

**Comment by qtdan** [comment]
@longxice please resolve your comment

**Review by qtdan — APPROVED** [review]
LGTM

**Comment by noradomi** [comment]
LGTM

**Comment by longxice** [comment]
LGTM

**Pull request merged**